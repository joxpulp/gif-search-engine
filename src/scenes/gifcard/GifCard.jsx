import {motion} from 'framer-motion';
import PropTypes from 'prop-types'

function GifCard({ gifurl, slug, onLoad, opacityAnimation, onClickImage }) {
	return (
		<motion.div onClick={onClickImage} initial={{opacity: 0}} animate={{opacity: opacityAnimation}} className="gif-card">
			<img className="gif-image" onLoad={onLoad} src={gifurl} alt={slug} />
		</motion.div>
	);
}

export default GifCard;

GifCard.propTypes = {
	gifurl: PropTypes.string.isRequired,
}