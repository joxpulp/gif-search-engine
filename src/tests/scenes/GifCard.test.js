import React from 'react'
import '@testing-library/jest-dom';
import {shallow} from 'enzyme'
import GifCard from '../../scenes/gifcard/GifCard'

describe('Pruebas componente <GifCard/>', () => {
    
    test('GifCard se renderiza correctamente', () => {

        const url= 'https://localhost/gif.jpg'
        const wrapper = shallow( <GifCard gifurl={url}/> )
        
        expect( wrapper ).toMatchSnapshot()
        
    })

}) 