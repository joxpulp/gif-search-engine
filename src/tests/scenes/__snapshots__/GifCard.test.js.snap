// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas componente <GifCard/> GifCard se renderiza correctamente 1`] = `
<ForwardRef(MotionComponent)
  animate={
    Object {
      "opacity": undefined,
    }
  }
  className="gif-card"
  initial={
    Object {
      "opacity": 0,
    }
  }
>
  <img
    className="gif-image"
    src="https://localhost/gif.jpg"
  />
</ForwardRef(MotionComponent)>
`;
